crates/parser/src/grammar/items.rs

```diff
@@ -414,10 +413,23 @@ fn fn_(p: &mut Parser<'_>, m: Marker) {
     m.complete(p, FN);
 }
 
-fn macro_call(p: &mut Parser<'_>) -> BlockLike {
+// test macro_rules_as_macro_name
+// macro_rules! {}
+// macro_rules! {};
+// macro_rules! ();
+// macro_rules! [];
+
+// test_err macro_rules_as_macro_name
+// macro_rules! ()
+// macro_rules! []
```

Comment:
These tests only test in item position, but not in expression position. For that we'd need something like
```rs
fn main() {
    let foo = macro_rules!();
}
```